### PR TITLE
Simplify chunk::Assert* and fix missing init check for async_future methods

### LINKED ISSUE #898: Some init asserts are missing in the generated code
As showed [here](https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/424#note_462625), some asserts are not generated whereas they should.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #898 

cc @sdroege @EPashkin

### CODE REVIEW

File: src/analysis/safety_assertion_mode.rs [hunk 1]
```diff
@@ -13,6 +14,16 @@ impl Default for SafetyAssertionMode {
     }
 }
 
+impl fmt::Display for SafetyAssertionMode {
```
Comment (EPashkin) [inline src/analysis/safety_assertion_mode.rs:17]: @sdroege Use Display for code snippets is acceptable for you?

**Comment by sdroege** [comment]
@GuillaumeGomez This is also for async functions, not only for async_future functions or not?

File: src/analysis/safety_assertion_mode.rs (hunk 1 above)
Comment (sdroege) [reply src/analysis/safety_assertion_mode.rs:17]: No, but your decision :)

Comment (EPashkin) [reply src/analysis/safety_assertion_mode.rs:17]: @GuillaumeGomez  can you move this match as function to writer?
IMHO just to "mod.rs" will do.

Comment (GuillaumeGomez) [reply src/analysis/safety_assertion_mode.rs:17]: Ok!

**Comment by GuillaumeGomez** [comment]
The async functions work as expected.

**Comment by sdroege** [comment]
> The async functions work as expected.

The futures functions don't need assertions because they just call the async functions anyway.

**Comment by GuillaumeGomez** [comment]
But the script doesn't care about it. Also, I'm not sure if it'd be a very nice idea to not check functions ending with `_future`. But your call.

**Comment by sdroege** [comment]
> But the script doesn't care about it. Also, I'm not sure if it'd be a very nice idea to not check functions ending with `_future`. But your call.

Could add the `skip` version of the macro to these functions

**Comment by GuillaumeGomez** [comment]
Fine by me!

**Comment by GuillaumeGomez** [comment]
Updated!

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks, waiting CI

**Pull request merged**